java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/SymbolicValue.java

```diff
@@ -201,13 +208,45 @@ public String toString() {
 
   public static class NotSymbolicValue extends UnarySymbolicValue {
 
+    public NotSymbolicValue() {
+      // used by ConstraintManager, operand is set by computedFrom
+    }
+
+    public NotSymbolicValue(SymbolicValue operand) {
+      super(operand);
+    }
 
     @Override
     public List<ProgramState> setConstraint(ProgramState programState, BooleanConstraint booleanConstraint) {
       return operand.setConstraint(programState, booleanConstraint.inverse());
     }
 
     @Override
+    protected List<ProgramState> copyAllConstraints(ProgramState programState) {
+      Preconditions.checkState(operand instanceof RelationalSymbolicValue,
+        "Copy of constraints should only be done on relational SV, but operand is %s", operand);
+      return ((RelationalSymbolicValue) operand).inverse().copyAllConstraints(programState);
+    }
+
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      NotSymbolicValue other = (NotSymbolicValue) o;
+      return operand.equals(other.operand);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(operand);
```

Comment:
This one is borderline:  I had to lookup doc to be sure hash will be different from operand... maybe a test to ensure the difference to document bad modification ?